VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts

```diff
@@ -458,25 +458,27 @@ export abstract class WaitingRoomBaseDirective {
 
         this.eventHubSubscription$.add(
             this.eventService.getParticipantsUpdated().subscribe(async participantsUpdatedMessage => {
-                this.logger.debug(`[WR] - Participant Updated`, participantsUpdatedMessage.participants);
+                this.logger.debug(`[WR] - Participants Updated`, participantsUpdatedMessage.participants);
                 const newParticipants = participantsUpdatedMessage.participants.filter(
                     x => !this.conference.participants.find(y => y.id === x.id)
                 );
                 newParticipants.forEach(participant => {
+                    this.logger.debug(`[WR] - Participant added, showing notification`, participant);
                     this.notificationToastrService.showParticipantAdded(
                         participant,
                         this.participant.status === ParticipantStatus.InHearing
                     );
                 });
 
                 const updatedParticipants = [...participantsUpdatedMessage.participants].map(updatedParticipant => {
-                    updatedParticipant.current_room = this.conference.participants.find(
-                        currentParticipant => currentParticipant.id === updatedParticipant.id
-                    )?.current_room;
+                    const currentParticipant = this.conference.participants.find(x => x.id === updatedParticipant.id);
+                    updatedParticipant.current_room = currentParticipant ? currentParticipant.current_room : null;
```

Comment:
if the current_room is null means couldn't get the room.label, I'm considering this will cause some css issue but I might wrong

Reply:
This is only for new participants who will not have a room at the start. Currently they will have an undefined room anyway but null evaluates a little faster(this is negligible but still there).

I actually would like to change the general structure of participants, as this model has everything, it would be nice for it to be split between details which don't change except via admin web and state such as room and status which change constantly via pexip/our own events.

Reply:
The updatedParticipant name might be a bit confusing here. I've changed it in a later commit.